jwst/pipeline/calwebb_spec2.py

```diff
@@ -295,7 +295,8 @@ def process_exposure_product(
             self.cube_build.suffix = 's3d'
             self.cube_build.save_results = False
             result_extra = self.cube_build(result)
-            self.save_model(result_extra[0], 's3d')
+            if self.cube_build.skip == False:
+                self.save_model(result_extra[0], 's3d')
```

Comment:
Actually, given that the suffix is set above, couldn't you just set
```
self.cube_build.save_results = True
```
and then if the step is skipped, that will get turned off?

Reply:
``save_results`` is intentionally turned off, for some reason, right before calling the step. I assume that's because the result needs to be saved in some special way that the ``cube_build`` step itself doesn't do properly? This is the only product explicitly saved by the pipeline module. All the rest are handled by ``stpipe``. Perhaps @jemorrison or @stscieisenhamer can shed more light on why we don't just let the step do the saving.

Reply:
There should be a difference with how cube_build save data and other steps in the cal_spec2 pipeline. One input product one output product and it is returned as an IFU cube model.

Reply:
All of the steps have one input and one output. Are you saying that ``cube_build`` does NOT have this 1 to 1 relationship? I know that for MIRI there are times where ``cube_build`` can create 2 output cubes from 1 input image (one for each of the 2 channel's worth of data in the image), but I thought that because ``calwebb_spec2`` explicitly sets
```
self.cube_build.output_type = 'multi'
```
before calling the step, that it would cause all of the data to be included in a single output cube. Hence it should have 1 input and 1 output in this case.

Reply:
Sorry  mistyped - My bad  I fixed the above typo . I am saying for the calspec2 pipeline cube_build has 1 output.

Reply:
Hmmm, so it looks like the explanation is in #1274, which I implemented. As explained in that PR, it's because ``cube_build`` has a lot of internal machinery to build complex output file names, but for use in ``calwebb_spec2`` all we want a simple exposure-based file name just like the input, with the suffix changed to "s3d". So the saving of the cube by the step was bypassed and moved up into the pipeline module.